experiments/experiment_prototype.py

```diff
@@ -1,385 +1,1111 @@
+        else:
+            exp_slice['rxonly'] = True
+            exp_slice['clrfrqflag'] = False
+
+    def check_slice_specific_requirements(self, exp_slice):
+        """
+        Check the requirements for the specific slice type as identified by the identifiers
+        rxonly and clrfrqflag. The keys that need to be checked depending on these identifiers 
+        are "txfreq", "rxfreq", and "clrfrqrange". This function may modify these keys.
+        :param exp_slice: 
+        :return: 
+        """
+        if exp_slice['clrfrqflag']:  # TX and RX mode with clear frequency search.
+            # In this mode, clrfrqrange is required along with the other requirements.
+            try:
+                assert isinstance(exp_slice['clrfrqrange'], list)
+                assert len(exp_slice['clrfrqrange']) == 2
+                assert isinstance(exp_slice['clrfrqrange'][0], int)
+                assert isinstance(exp_slice['clrfrqrange'][1], int)
+            except AssertionError:
+                errmsg = 'clrfrqrange must be an integer list of length = 2'
+                raise ExperimentException(errmsg)
+            try:
+                assert exp_slice['clrfrqrange'][0] < exp_slice['clrfrqrange'][1]
+                assert (exp_slice['clrfrqrange'][1] * 1000) < self.options.max_freq
+                assert (exp_slice['clrfrqrange'][0] * 1000) > self.options.min_freq
+            except AssertionError:
+                errmsg = """clrfrqrange must be between min and max frequencies for the radar
+                            and must have lower frequency first.
+                            """
+                raise ExperimentException(errmsg)
+
+            still_checking = True
+            while still_checking:
+                for freq_range in self.options.restricted_ranges:
+                    try:
+                        assert exp_slice['clrfrqrange'][0] not in range(freq_range[0],
+                                                                        freq_range[1])
+                    except AssertionError:
+                        try:
+                            assert exp_slice['clrfrqrange'][1] not in range(freq_range[0],
+                                                                            freq_range[1])
+                        except AssertionError:
+                            # the range is entirely within the restricted range.
+                            raise ExperimentException('clrfrqrange is entirely within restricted '
+                                                      'range {}'.format(freq_range))
+                        else:
+                            # TODO Log warning, changing clrfrqrange because lower portion is in a
+                            # restricted frequency range.
+                            exp_slice['clrfrqrange'][0] = freq_range[
+                                                              1] + 1  # outside of restricted range.
```

Comment:
`int FreqTest(struct FreqTable *ptr,int freq) { 
  int i;
  if ((freq<8000) || (freq>20000)) return 1;
  if (ptr==0) return 0;
  if (ptr->num==0) return 0;

  for (i=0;i<ptr->num;i++) 
    if ((ptr->start[i]<=freq) && (freq<=ptr->end[i])) return 1;

  return 0;  
}
`

This is in the freq.1.01 library. It is not called in programs with clear frequency search from what I can see (twofsound, for example). It is called in epopsound.c.